code/modules/implant_hacking/_hacking.dm

```diff
@@ -0,0 +1,162 @@
+/obj/item/cyberlink_connector
+	name = "Wireless personal connector"
+	desc = "Allows you to connect to incompatible implants and hack them."
+	icon = 'icons/obj/surgery.dmi'
+	icon_state = "connector"
+	var/obj/item/organ/cyberimp/cyberlink/parent_cyberlink
+	var/list/datum/hacking_minigame/game_list = list()
+	var/time_left = 0
+	var/obj/item/organ/cyberimp/cybernetic
+	var/mob/living/current_user
+	var/processing = FALSE
+
+/obj/item/cyberlink_connector/Destroy()
+	. = ..()
+	parent_cyberlink = null
+	cleanup()
+	STOP_PROCESSING(SSprocessing,src)
+	processing = FALSE
+
+///We dont open the tgui when we click on this.
+/obj/item/cyberlink_connector/interact(mob/user)
+	add_fingerprint(user)
+
+/obj/item/cyberlink_connector/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	. = ..()
+	if(!proximity_flag)
+		return
+
+	if(!istype(target,/obj/item/organ/cyberimp) || istype(target,/obj/item/organ/cyberimp/cyberlink))
+		return
+
+	if(!parent_cyberlink)
+		var/obj/item/organ/cyberimp/cyberlink/link = user.getorganslot(ORGAN_SLOT_LINK)
+		if(!link)
+			to_chat(user,"<span class='notice'> NO CYBERLINK DETECTED </span>")
+			return
+		parent_cyberlink = link
+
+	game_list = list()
+
+	var/diffrences = 3
+	current_user = user
+	cybernetic = target
+
+	for(var/info in cybernetic.encode_info)
+		if(cybernetic.encode_info[info] == 0)
+			continue
+		var/list/encrypted_information = cybernetic.encode_info[info]
+
+		for(var/protocol in encrypted_information)
+			if(protocol in parent_cyberlink.encode_info[info])
+				diffrences--
+				break
+
+	if(diffrences == 0)
+		to_chat(current_user,"<span class='notice'> Cyberlink beeps: [uppertext(cybernetic.name)] ALREADY COMPATIBLE.</span>")
+		cleanup()
+		return
+
+	if(!game_list.len)
+		for(var/i in 1 to diffrences)
+			var/datum/hacking_minigame/game = new/datum/hacking_minigame()
+			game.generate()
+			game_list += game
+
+	ui_interact(user)
+
+/obj/item/cyberlink_connector/proc/cleanup()
+	current_user = null
+	cybernetic = null
+	QDEL_LIST(game_list)
+	processing = FALSE
+
+/obj/item/cyberlink_connector/proc/hack_success(success as num)
+	for(var/info in cybernetic.encode_info)
+		if(cybernetic.encode_info[info] == 0)
+			continue
```

Comment:
what

Reply:
when cybernetic encode protocol is equal to 0 it is meant to signify it is compatible with everything, meaning we don't need to check it against the link encode protcols, since by definition, it is compatible with everything.

Reply:
Would this be better suited to defines? or do the protocalls hold  info about how much stuff  there is left to use and such

Reply:
well all encode info is already made into define, just the 0 isn't and frankly it should be an easy thing to change.